From 61cb52c1653b742a79fb1db5b647be80b9d14e77 Mon Sep 17 00:00:00 2001
From: "Daisuke Fujimura (fd0)" <booleanlabel@gmail.com>
Date: Mon, 23 Sep 2019 11:42:42 +0200
Subject: [PATCH] Provide a system-wide pkgconfig file

Configure, build/create and install the 'libdsk.pc' file.

YACP - Yet Another Cygwin Ports (https://github.com/fd00/yacp)

Originated-by: [YACP] libdsk-1.5.12-1bl1.src.patch
Originated-md5: 2bfb00203a436ea2dea04f4813d4eecc
Originated-URL: https://github.com/fd00/yacp/raw/37e7155/libdsk/libdsk-1.5.12-1bl1.src.patch

Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 Makefile.am     |  3 ++-
 configure.ac    |  2 +-
 lib/Makefile.am |  2 +-
 libdsk.pc.in    | 10 ++++++++++
 4 files changed, 14 insertions(+), 3 deletions(-)
 create mode 100644 libdsk.pc.in

diff --git a/Makefile.am b/Makefile.am
index a71befe..17269b5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -54,4 +54,5 @@ EXTRA_DIST= $(srcdir)/config/* \
 install-man:
 	cd man && make install-man
 
-
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(PACKAGE).pc
diff --git a/configure.ac b/configure.ac
index 49f7387..724b851 100644
--- a/configure.ac
+++ b/configure.ac
@@ -128,4 +128,4 @@ AC_CHECK_FUNCS(strerror)
 AC_SUBST(LIBDSKJAR)
 AC_SUBST(TOOLCLASSES)
 AC_SUBST(CLASSPATH)
-AC_OUTPUT(Makefile doc/Makefile include/Makefile lib/Makefile tools/Makefile man/Makefile)
+AC_OUTPUT(libdsk.pc Makefile doc/Makefile include/Makefile lib/Makefile tools/Makefile man/Makefile)
diff --git a/lib/Makefile.am b/lib/Makefile.am
index bdfd610..759503a 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -30,7 +30,7 @@ EXTRA_DIST=$(CLASSDPRE)/Drive.java \
 # If interfaces have been removed set the right-hand number to 0.
 # 
 #
-libdsk_la_LDFLAGS = -version-info 8:1:3
+libdsk_la_LDFLAGS = -no-undefined -version-info 8:1:3
 libdsk_la_SOURCES = dskread.c  dskwrite.c dskcheck.c dskstat.c \
 		   dsklphys.c dskfmt.c   dskopen.c  dskpars.c \
 		   dskerror.c dskseek.c  dsksecid.c dskgeom.c \
diff --git a/libdsk.pc.in b/libdsk.pc.in
new file mode 100644
index 0000000..328dc24
--- /dev/null
+++ b/libdsk.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libdsk
+Description: Library for accessing discs and disc image files
+Version: @VERSION@
+Libs: -L${libdir} -ldsk
+Cflags: -I${includedir}
